### PR TITLE
feat: add manual-revocation-check script

### PR DESCRIPTION
**Pull request by raleng**
# Description

This adds a script to demonstrate how to retrieve the revocation list from the public endpoint, how to decode the list, and how to check the credential status.

For convenience, this also let's you pass an encoded compact or compact-semantic credential. The script will decode the credential to retrieve the revocation list URL and index.

### CODE REVIEW

File: manual-revocation-check/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+[![MATTR](../docs/assets/mattr-logo-square.svg)](https://github.com/mattrglobal)
+
+# Manual revocation check
+
+## Overview
+
+Credential revocation is a critical feature that allows issuers to invalidate previously issued credentials, ensuring the integrity and security of the system.
+
+MATTR provides capabilities to issue revokable credentials, to manage the revocation status of credentials, and to check the revocation status of a credential. Check out our [Credential Management](https://learn.mattr.global/docs/capabilities/management) capabilities and our [Revocation Tutorial](https://learn.mattr.global/tutorials/management/revocation) to learn more.
```
Comment (avner-m) [inline manual-revocation-check/README.md:9]:
```suggestion
MATTR provides capabilities to issue revokable credentials, to manage the revocation status of credentials, and to check the revocation status of a credential. Check out our [credential management](https://learn.mattr.global/docs/capabilities/management) capabilities and our [Revocation tutorial](https://learn.mattr.global/tutorials/management/revocation) to learn more.
```

File: manual-revocation-check/README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+While we highly suggest to leverage your MATTR tenant for credential management, credentials also include the necessary information to check their revocation status manually to enable interoperability.
```
Comment (avner-m) [inline manual-revocation-check/README.md:11]:
```suggestion
While we highly recommend to leverage your MATTR tenant for credential management, credentials also enable unauthenticated relaying parties to check their revocation status to enable interoperability.
```

File: manual-revocation-check/README.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+This sample script demonstrates how to retrieve the revocation list data, how to decode the data, and how to check the revocation status.
+
+## How to run the script
+
+### Web Semantic Credentials
```
Comment (avner-m) [inline manual-revocation-check/README.md:17]:
```suggestion
### JSON credentials
```

File: manual-revocation-check/README.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+Web Semantic credentials contain a `credentialStatus` field that holds the reference to a revocation list.
```
Comment (avner-m) [inline manual-revocation-check/README.md:19]:
```suggestion
JSON credentials contain a `credentialStatus` field that holds the reference to a revocation list.
```

File: manual-revocation-check/README.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+```
+
+You can check the revocation status of this credential by running the script with node, providing the `revocationListCredential` and `revocationListIndex` as arguments.
+
+
+```bash
+node manual-revocation-check.js https://learn.vii.au01.mattr.global/core/v2/credentials/web-semantic/revocation-lists/0ec79c8e-9859-46c0-a277-6e48f468b16e 1
+```
+
+### Compact and Compact-Semantic Credentials
```
Comment (avner-m) [inline manual-revocation-check/README.md:37]:
```suggestion
### CWT and semantic CWT credentials
```

File: manual-revocation-check/README.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+Compact and compact-semantic credentials are usually shared and verified in their encoded forms. For convenience, you can provide the encoded credential or the revocation list URL and index to the script.
```
Comment (avner-m) [inline manual-revocation-check/README.md:39]:
```suggestion
CWT and Semantic CWT credentials are usually shared and verified in their encoded forms. For convenience, you can provide the encoded credential or the revocation list URL and index to the script.
```

File: manual-revocation-check/README.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+If you provide the encoded compact credential, the script will decode the credential and retrieve the list URL and index.
```
Comment (avner-m) [inline manual-revocation-check/README.md:41]:
```suggestion
If you provide the encoded form, the script will decode the credential and retrieve the list URL and index.
```

File: manual-revocation-check/README.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+```bash
+node manual-revocation-check.js CSC:/1/2KCE3I...
+```
+
+If you decode the compact credential yourself, you can pass the revocation list URL and index.
```
Comment (avner-m) [inline manual-revocation-check/README.md:47]:
```suggestion
If you decode the credential yourself, you can pass the revocation list URL and index.
```

File: manual-revocation-check/manual-revocation-check.js [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+    let bits = byte.toString(2);
+    for (let bit of bits) {
+      bitArray.push(Number(bit));
+    }
+  }
+
+  return bitArray;
+}
+
+// Compact Credentials are base32 encoded CBOR web tokens
```
Comment (avner-m) [inline manual-revocation-check/manual-revocation-check.js:34]:
```suggestion
// CWT credentials are base32 encoded CBOR web tokens
```

File: manual-revocation-check/manual-revocation-check.js [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+// Revocable compact credentials contain a reference to the
```
Comment (avner-m) [inline manual-revocation-check/manual-revocation-check.js:35]:
```suggestion
// Revocable CWT credentials contain a reference to the
```

File: manual-revocation-check/manual-revocation-check.js [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+  const payload = cborDecode(cwt.value[2]);
+
+  const status = payload["-65537"];
+  const index = status["2"];
+  const listUrl = status["3"];
+
+  return { listUrl, index };
+}
+
+// The public revocation list endpoint for compact credentials
```
Comment (avner-m) [inline manual-revocation-check/manual-revocation-check.js:49]:
```suggestion
// The public revocation list endpoint for CWT credentials
```

File: manual-revocation-check/manual-revocation-check.js [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+  const payload = cborDecode(cwt.value[2]);
+
+  const compressedList = payload["-65538"];
+  const list = pako.ungzip(compressedList);
+  const revocationList = toBitArray(list);
+
+  return revocationList[index] === 1;
+}
+
+// The public revocation list endpoint for web semantic credentials
```
Comment (avner-m) [inline manual-revocation-check/manual-revocation-check.js:66]:
```suggestion
// The public revocation list endpoint for JSON credentials
```

File: manual-revocation-check/README.md
Comment (avner-m) [inline manual-revocation-check/README.md:1]: Do you think we could/should add an example credential of each type for them to play with?

Also do we want to show what the result of that script looks like or explain what it actually does?

**Review by avner-m — CHANGES_REQUESTED** [review]

File: manual-revocation-check/README.md (hunk 2 above)
Comment (raleng) [reply manual-revocation-check/README.md:11]:
```suggestion
While we highly recommend to leverage your MATTR tenant for credential management, credentials also enable unauthenticated relying parties to check their revocation status to enable interoperability.
```

File: manual-revocation-check/README.md
Comment (raleng) [reply manual-revocation-check/README.md:1]: A JSON credential example is basically there already (though we only share the `credentialStatus` object, not the whole credential). We certainly can add a compact credential as well.

The output is just text in the terminal, but I can add a section to give a sample output.

**Review by preetpatel — APPROVED** [review]

File: manual-revocation-check/README.md [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+```
+
+You can check the revocation status of this credential by running the script with node, providing the `revocationListCredential` and `revocationListIndex` as arguments.
+
+
+```bash
+node manual-revocation-check.js https://learn.vii.au01.mattr.global/core/v2/credentials/web-semantic/revocation-lists/0ec79c8e-9859-46c0-a277-6e48f468b16e 1
+```
+
+### CWT and semantic CWT credentials
```
Comment (avner-m) [inline manual-revocation-check/README.md:37]:
```suggestion
### CWT and Semantic CWT credentials
```

**Review by avner-m — APPROVED** [review]

**Pull request merged**